### ISSUE TITLE
3.0b3/2.0 w/macOS13 - Issue w/Super waiting for download but download never starts

### ISSUE THREAD

**Issue by iDrewbs** [Macjutsu/super#43]
I've been running into an issue where super is going through the MDM download process, gets to the point where it's waiting for the download to start, the MDM log show's the MDM download workflow as complete, but the download isn't starting. I've had it try probably 20+ attempts, tried a reboot, also tried version 2.0 but had the same resut
This is running on macOS 13.0 trying to update to 13.1 beta so I'm not sure if that's causing an issue, but wanted to call it out and send over the logs in case there is an issue with softwareupdate on macOS 13 or if you spot something I'm missing

**[super (1).log](https://github.com/Macjutsu/super/files/9957036/super.1.log)
[mdm (1).log](https://github.com/Macjutsu/super/files/9957037/mdm.1.log)**
**[update (1).log](https://github.com/Macjutsu/super/files/9957678/update.1.log)**

**Comment by Macjutsu** [comment]
This is not an issue with `super`. After the Jamf API command is sent, there is nothing `super` can do but wait for the download to start. We literally have no control over the download process after the MDM command workflow is started... none. Instead `super` waits for `softwareupdated` to start the download to complete, and quite often it simply will not. The update.log is the raw output of the `softwareupdated` process, and if it never starts to download then `super` v3.0 will properly timeout and try again. See [this example log](https://github.com/Macjutsu/super/blob/main/Example-Logs/update.log) for what should happen when the download is working.

**Comment by Macjutsu** [comment]
Reopening for clarification... are the affected systems set to automatically download updates? If so, that could be the cause. In other words, if the update is already downloaded before the MDM command tells it to download, then the `softwareupdate` log may not indicate any download progress because it's already done. If this is the case then there could be something `super` could do to try and detect if the update was previously downloaded automatically.

**Comment by Macjutsu** [comment]
The problem is that there aren't any good methods to detect this. The only "official" method is to literally run `softwareupdate --download` every time `super` runs to validate previous downloads, but as we all now know... every run of `softwareupdate` is another chance for it to hang. Normally `super` keeps track of downloads it completes to avoid re-downloading, so did you reset `super` between download/install attempts?

**Comment by iDrewbs** [comment]
Yes I did reset `super` between download/install attempts but it didn't have any effect. 
I have a feeling this would be a rare occurrence. It's only come up this once, and `super` has been able to reset other times. 
Maybe in the future a flag could be added to force a re-download of all updates but it's probably a low priority at this point.

**Comment by iDrewbs** [comment]
It looks like someone submitted a pull request for what sounds like this same issue, I haven't had a chance to comb over their code though

**Comment by Macjutsu** [comment]
It's not ideal as it still requires running softwareupdate again... an ideal solution would be to know before any download is attempted if the update is already cached. I found this... working on it... `defaults read /System/Volumes/Update/Update.plist update-asset-attributes`

**Comment by Macjutsu** [comment]
Well.. I've tested some things... I will likely be re-architecting the download mechanism again... stay tuned for the next beta.
